packages/hub/tasks/notify-prepaid-card-drop.ts

```diff
@@ -0,0 +1,50 @@
+import { inject } from '@cardstack/di';
+import config from 'config';
+import NotificationPreferenceService from '../services/push-notifications/preferences';
+import WorkerClient from '../services/worker-client';
+import { PushNotificationData } from './send-notifications';
+import { generateContractEventNotificationId } from '../utils/notifications';
+
+const web3Config = config.get('web3') as { layer2Network: 'sokol' | 'xdai' };
+
+export default class NotifyPrepaidCardDrop {
+  notificationPreferenceService: NotificationPreferenceService = inject('notification-preference-service', {
+    as: 'notificationPreferenceService',
+  });
+  workerClient: WorkerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform({ transactionHash, ownerAddress }: { transactionHash: string; ownerAddress: string }) {
```

Comment:
This function signature is tragically broken! I don’t know why I thought I could do this… the event listener only passes `transactionHash` as a bare string.

But [here](https://github.com/cardstack/cardstack/commit/a9aa73e61a2efb9dcff23cc22c7091d9f6bd10bf) I am experimenting with passing the entire event. I think that will help because otherwise we don’t have access to the owner address needed to send the notification, and it does come with the event.

Here’s an event preserved for future reference:

```
{
  "address": "0xb52726CbDA92b5340F32A0e837e8678E68Ff9da8",
  "blockHash": "0x53f3b77bac7cdcb7f952ab9f705f405700d1edc769044343f6e8abe3c735db5f",
  "blockNumber": 26785687,
  "logIndex": 0,
  "removed": false,
  "transactionHash": "0xbaf65d22b29c635243a2e5fa748eb195577c0a7bd342081dc5907c488323e25c",
  "transactionIndex": 23,
  "transactionLogIndex": "0x0",
  "id": "log_e3b8e103",
  "returnValues": {
    "0": "0x323B2318F35c6b31113342830204335Dac715AA8",
    "1": "0x905b1faa1a56a0fd8447a469f8aa558665d550e56d60d106cd1096c526df4bdd",
    "owner": "0x323B2318F35c6b31113342830204335Dac715AA8",
    "sku": "0x905b1faa1a56a0fd8447a469f8aa558665d550e56d60d106cd1096c526df4bdd"
  },
  "event": "PrepaidCardProvisioned",
  "signature": "0x7b8bf0fde69780a9f1eef765d847919a85bff2d761a03e306d0ce4f884c0f7c3",
  "raw": {
    "data": "0x000000000000000000000000323b2318f35c6b31113342830204335dac715aa8905b1faa1a56a0fd8447a469f8aa558665d550e56d60d106cd1096c526df4bdd",
    "topics": [
      "0x7b8bf0fde69780a9f1eef765d847919a85bff2d761a03e306d0ce4f884c0f7c3"
    ]
  }
}
```

Reply:
Oops, I missed that too. It is a bit sad that Typescript doesn't catch these at the moment.